src/utils/colorManipulation.ts

```diff
@@ -13,20 +13,81 @@ export function computeBacklightFrame(
   videoDimensions: Dimensions,
   opts: BacklightOptions
 ) {
+  const newFrame = new Uint8ClampedArray(frame.data.length);
   const divisions = computeDivisions(
     frame,
     videoDimensions,
     opts.horizontalDivisions,
     opts.verticalDivisions,
     false
   );
+  const oneFourthDivWidth = Math.round(divisions[0].width / 4);
+  const oneFourthDivHeight = Math.round(divisions[0].height / 4);
 
-  divisions.forEach((division) => {
-    ctx.fillStyle = `rgb(${division.color[RED_CHANNEL_OFFSET]}, ${division.color[GREEN_CHANNEL_OFFSET]}, ${division.color[BLUE_CHANNEL_OFFSET]})`;
-    ctx.fillRect(division.col, division.row, division.width, division.height);
-  });
+  for(let div = 0; div < divisions.length; div += 2) {
+    const currDiv = divisions[div];
+    const nextDiv = divisions[div + 1];
 
-  return frame;
+    // setting all the divs to hav the averaged colors
+    for (let row = currDiv.row; row < currDiv.row + currDiv.height; row++) {
+      for (let col = currDiv.col; col < currDiv.col + currDiv.width; col++) {
+        const frameIndex = (row * frame.width + col) * 4;
+
+        newFrame[frameIndex + RED_CHANNEL_OFFSET] = currDiv.color[RED_CHANNEL_OFFSET];
+        newFrame[frameIndex + GREEN_CHANNEL_OFFSET] = currDiv.color[GREEN_CHANNEL_OFFSET];
+        newFrame[frameIndex + BLUE_CHANNEL_OFFSET] = currDiv.color[BLUE_CHANNEL_OFFSET];
+        newFrame[frameIndex + ALPHA_CHANNEL_OFFSET] = currDiv.color[ALPHA_CHANNEL_OFFSET];
+      }
+    }
+
+    // set the nextDiv to have its respective average color
+    for (let row = nextDiv.row; row < nextDiv.row + nextDiv.height; row++) {
+      for (let col = nextDiv.col; col < nextDiv.col + nextDiv.width; col++) {
+        const frameIndex = (row * frame.width + col) * 4;
+
+        newFrame[frameIndex + RED_CHANNEL_OFFSET] = nextDiv.color[RED_CHANNEL_OFFSET];
+        newFrame[frameIndex + GREEN_CHANNEL_OFFSET] = nextDiv.color[GREEN_CHANNEL_OFFSET];
+        newFrame[frameIndex + BLUE_CHANNEL_OFFSET] = nextDiv.color[BLUE_CHANNEL_OFFSET];
+        newFrame[frameIndex + ALPHA_CHANNEL_OFFSET] = nextDiv.color[ALPHA_CHANNEL_OFFSET];
+      }
+    }
+    
+    // attempting to call regionConvolution to the right at the top row
+    if(!(nextDiv.col + nextDiv.width >= frame.width)) {
+      regionConvolution(frame, newFrame, currDiv.row, nextDiv.col - oneFourthDivWidth, currDiv.row + currDiv.height, nextDiv.col + oneFourthDivWidth, 3);
+    }
+
+  }
+  /**
+   * SECOND POSSIBLE METHOD FOR CALLING REGIONCONVOLUTION:
+   * after coloring all of the divisions to their respective average colors,
+   * we can call region convolution on ALL divisons in the DOWN and RIGHT
+   * directions UNLESS they are touching the right or bottom side, if they do 
+   * then we can call it either just going down or to the right respectively.
```

Comment:
no strong feelings here, both approaches are great and creative. go with whatever you think will work best and is the most readable